src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -75,10 +77,16 @@ def get(url, headers)
     request_options = RequestOptions.new(build_url(url))
     request_options.set_headers(headers)
     request_options.set_as(ResponseBodyType::TEXT)
-    response = @client.get(request_options)
+    response = @@client.get(request_options)
```

Comment:
same here; this would probably be better as `self.client_get`, or just `self.client.get`.  The latter is probably fine and then you could skip the step I suggested above about defining a `self.client_post` method, and just use `self.client`.